api/src/services/items.ts

```diff
@@ -140,58 +140,56 @@ export class ItemsService<Item extends AnyItem = AnyItem, Collection extends str
 		const payload: AnyItem = cloneDeep(data);
 		const nestedActionEvents: ActionEventParams[] = [];
 
-		// By wrapping the logic in a transaction, we make sure we automatically roll back all the
-		// changes in the DB if any of the parts contained within throws an error. This also means
-		// that any errors thrown in any nested relational changes will bubble up and cancel the whole
-		// update tree
-		const primaryKey: PrimaryKey = await transaction(this.knex, async (trx) => {
-			const serviceOptions: AbstractServiceOptions = {
-				accountability: this.accountability,
-				knex: trx,
-				schema: this.schema,
-			};
-
-			// We're creating new services instances so they can use the transaction as their Knex interface
-			const payloadService = new PayloadService(this.collection, serviceOptions);
-
-			// Run all hooks that are attached to this event so the end user has the chance to augment the
-			// item that is about to be saved
-			const payloadAfterHooks =
-				opts.emitEvents !== false
-					? await emitter.emitFilter(
-							this.eventScope === 'items'
-								? ['items.create', `${this.collection}.items.create`]
-								: `${this.eventScope}.create`,
-							payload,
-							{
-								collection: this.collection,
-							},
-							{
-								database: trx,
-								schema: this.schema,
-								accountability: this.accountability,
-							},
-					  )
-					: payload;
-
-			const payloadWithPresets = this.accountability
-				? await processPayload(
+		// Run all hooks that are attached to this event so the end user has the chance to augment the
+		// item that is about to be saved
+		const payloadAfterHooks =
+			opts.emitEvents !== false
+				? await emitter.emitFilter(
+						this.eventScope === 'items'
+							? ['items.create', `${this.collection}.items.create`]
+							: `${this.eventScope}.create`,
+						payload,
 						{
-							accountability: this.accountability,
-							action: 'create',
 							collection: this.collection,
-							payload: payloadAfterHooks,
 						},
 						{
-							knex: trx,
+							database: this.knex,
```

Comment:
Should this event callback not stay within in the transaction so extensions get the knex transaction and can potentially cancel/rollback.

Reply:
hmmm, as discussed it should and will move it back. The original thinking was to provide access to `payloadWithPresets` without needing an alternate variable assigned inside the transaction

Reply:
This should be resolved via 866fabc7a886df1a1fd2d4ba30c60c567117835d